From 767bfd666f142dcfe2cd6401e0d5e040b3c7004d Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Wed, 9 May 2012 16:33:44 +0300
Subject: [PATCH 158/311] conf/obj.h: remove upward relations

These pointers to "upper" configuration objects may be introduced
later, when the need arises.
---
 conf/obj.h | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/conf/obj.h b/conf/obj.h
index 62b4195..88dfdb9 100644
--- a/conf/obj.h
+++ b/conf/obj.h
@@ -131,7 +131,7 @@ enum c2_conf_objtype {
  * configuration object.
  */
 struct c2_conf_obj {
-	enum c2_conf_objtype co_type;
+	enum c2_conf_objtype co_type; /* XXX try to get rid of this field */
 	struct c2_conf_buf   co_key;
 	enum c2_conf_status  co_status;
 	/**
@@ -172,16 +172,15 @@ struct c2_conf_filesystem {
 struct c2_conf_node;
 
 struct c2_conf_service {
-	struct c2_conf_obj         cs_obj;
-	struct c2_conf_filesystem *cs_filesystem;
-	enum c2_cfg_service_type   cs_type;
+	struct c2_conf_obj       cs_obj;
+	enum c2_cfg_service_type cs_type;
 	char cs_endpoints[C2_CFG_SERVICE_MAX_END_POINTS][C2_CFG_NAME_LEN];
 /* private: internal use only */
-	struct c2_conf_node       *cs_node;
+	struct c2_conf_node     *cs_node;
 	/** Linkage to c2_conf_filesystem::cf_services. */
-	struct c2_tlink            cs_filesystem_linkage;
+	struct c2_tlink          cs_filesystem_linkage;
 	/** Magic for a list of services. */
-	uint64_t                   cs_magic;
+	uint64_t                 cs_magic;
 };
 
 /** Descriptor for a tlist of filesystem's services. */
@@ -202,7 +201,6 @@ struct c2_conf_node {
 
 struct c2_conf_nic {
 	struct c2_conf_obj   ni_obj;
-	struct c2_conf_node *ni_node;
 	uint32_t             ni_iface_type;
 	uint32_t             ni_mtu;
 	uint64_t             ni_speed;
@@ -212,7 +210,6 @@ struct c2_conf_nic {
 
 struct c2_conf_sdev {
 	struct c2_conf_obj   sd_obj;
-	struct c2_conf_node *sd_node;
 	uint32_t             sd_type;
 	uint32_t             sd_media;
 	uint64_t             sd_size;
@@ -225,7 +222,6 @@ struct c2_conf_sdev {
 
 struct c2_conf_partition {
 	struct c2_conf_obj   pa_obj;
-	struct c2_conf_sdev *pa_sdev;
 	uint64_t             pa_start;
 	uint64_t             pa_size;
 	uint32_t             pa_index;
-- 
1.8.3.2

